ONSdigital/rm-actionexporter-service
# Sts008 receive notification letter

# Motivation and Context
This changes adds the template for a social notification letter CSV template and the mapping for social action requests. This enables the action exporter service to generate the CSV files containing the addresses and other details required to generate the social notification letters. These CSV files are sent to an SFTP server to be picked up downstream.

# What has changed
* Added social notification file templates and template mapping rows to the SQL changes and updated changelogs accordingly.
* Added a docker-compose file to start up the services required for an integration test in docker locally. These run with names and ports specific to this service so should not conflict with the containers in [ras-rm-docker-dev](https://github.com/ONSdigital/ras-rm-docker-dev).
* Added an integration test which ensures that when an action instruction for a social notification is put on the relevant queue, this service picks it up, generates the correct CSV file and then sends it to the SFTP. 


# How to test?
Running a `mvn clean install` should now run the integration test for these changes, including running the required over services in docker and running this service locally.


# Links
Related to: https://github.com/ONSdigital/rm-action-service/pull/48
Trello card: [Dev task 5: INT: Notify respondent of invitation to a survey (5)](https://trello.com/c/xbFrDWWf/100-dev-task-5-int-notify-respondent-of-invitation-to-a-survey-5)
